tests/baselines/reference/convertToAsyncFunction/convertToAsyncFunction_noArgs1.ts

```diff
@@ -26,5 +26,6 @@ async function main2() {
     console.log(".");
     await delay(500);
     console.log(".");
-    return delay(500);
+    return await delay(500);
```

Comment:
I could be wrong, but I believe we only need this change if the statement is inside a `try` block. Does that seem right, and do you think it would be difficult to limit the change to just those cases?

Reply:
Do you mean for the following code the need to omit `await` in return statement and add it only if if inside try block?
```ts
function main2() {
    console.log("Please wait. Loading.");
    return delay(500)
        .then(() => { console.log("."); return delay(500); })
        .then(() => { console.log("."); return delay(500); })
        .then(() => { console.log("."); return delay(500); })
}
```

Reply:
Yeah, only if the resulting generated code is inside a `try`. `return await` is usually regarded as poor style, kind of like returning `Promise.resolve(something)` in a `then`. But it has importantly different semantics when inside a `try` block because it makes the difference between the error being handled in the function implementation vs. propagating the rejection onto the caller.

Reply:
Example:

```ts
async function f1() {
  return await fetch('/'); // `await` is not useful here
}

async function f2() {
  try {
    // `await` is useful here; without it, the catch block would never execute.
    // f2 returns a Promise that never rejects. f1 returns a Promise that resolves
    // or rejects with `fetch('/')`, with or without the `await`.
    return await fetch('/');
  } catch {
    return {};
  }
}
```

Reply:
Thanks for the clarification.

Reply:
@andrewbranch Should these examples also omit `await` in `return`?

https://github.com/microsoft/TypeScript/blob/083129f005ac0e40b96701c7928434e8aee40655/tests/baselines/reference/convertToAsyncFunction/convertToAsyncFunction_nestedPromises.ts#L10-L12

or 

https://github.com/microsoft/TypeScript/blob/083129f005ac0e40b96701c7928434e8aee40655/tests/baselines/reference/convertToAsyncFunction/convertToAsyncFunction_callbackReturnsPromiseLastInChain.ts#L9-L11

I'm thinking about how to handle `await` inside/outside `try` statement. Maybe you have some thoughts about that?

Reply:
Hmm, that baseline pointed me to this PR: https://github.com/microsoft/TypeScript/pull/27573

I guess this is already a known limitation of the refactor. In that case, I’m ok with your PR as-is.

Reply:
Well, I’m a little confused as to why #27573 didn’t already apply to some of the baselines that changed in this PR. I do think that if we start adding a lot more `return await`, people are going to file issues about it. 🤔

Reply:
@andrewbranch Thanks for sharing this PR. I didn't know about it.

Reply:
Basically, this PR fixes a really simple issue, it just handles `return` statements in `{}`. For instance, two examples with the same logic behave differently. 

```ts
function fn(): Promise<void> {
    return Promise.resolve();
}

// Before
function f1() {
    return fn().then(() => fn());
}
// After 
async function f1() {
    await fn();
return await fn(); // await is added
}

// Before
function f2() {
    return fn().then(() => { return fn() });
}
// After 
async function f2() {
    await fn();
return fn(); // await is missing 
}
```
[Example ](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMMAKAlALkQBQE5wC2MAzgKYA8AbnDACYB8iA3gFCIeJ5lQh5L4ipMgDpuJOABsqZDAG5WAX1ahIsBMgCMGFu07de-ZGnQioACzJodAXiYoM6BctXR4SYACYdbTlx58HiZmltboiHYs-oZBOopOSkA
)


About for the test, there is a test for `{}`, however, there is no test for `()`. I decided to add, to handle both cases. https://github.com/microsoft/TypeScript/pull/39649/files#diff-71c285c7a401ed150a7e8367a2d6681cR1408-R1422